lightningd/peer_htlcs.c

```diff
@@ -637,7 +650,68 @@ static struct htlc_accepted_hook_response *
 htlc_accepted_hook_deserialize(const tal_t *ctx, const char *buffer,
 			       const jsmntok_t *toks)
 {
-	return NULL;
+	struct htlc_accepted_hook_response *response;
+	const jsmntok_t *resulttok, *failcodetok, *paykeytok, *chanupdtok;
+
+	if (!toks || !buffer)
+		return NULL;
+
+	resulttok = json_get_member(buffer, toks, "result");
+
+	/* If the result is "continue" we can just return NULL since
+	 * this is the default behavior for this hook anyway */
+	if (!resulttok) {
+		fatal("Plugin return value does not contain 'result' key %s",
+		      json_tok_full(buffer, toks));
```

Comment:
You want json_tok_strdup() here I think: json_tok_full just returns the pointer to the start of the token, which may not even be NUL terminated AFAIK.